redisvl/schema/fields.py

```diff
@@ -0,0 +1,115 @@
+from typing import Any, Dict, Optional
+from typing_extensions import Literal
+
+from pydantic import BaseModel, Field, validator
+from redis.commands.search.field import (
+    GeoField,
+    NumericField,
+    TagField,
+    TextField,
+    VectorField,
+)
+
+
+class BaseField(BaseModel):
+    name: str = Field(...)
+    sortable: Optional[bool] = False
+    as_name: Optional[str] = None
+
+
+class TextFieldSchema(BaseField):
+    weight: Optional[float] = 1
+    no_stem: Optional[bool] = False
+    phonetic_matcher: Optional[str] = None
+    withsuffixtrie: Optional[bool] = False
+
+    def as_field(self):
+        return TextField(
+            self.name,
+            weight=self.weight,
+            no_stem=self.no_stem,
+            phonetic_matcher=self.phonetic_matcher,
+            sortable=self.sortable,
+            as_name=self.as_name,
+        )
+
+
+class TagFieldSchema(BaseField):
+    separator: Optional[str] = ","
+    case_sensitive: Optional[bool] = False
+
+    def as_field(self):
+        return TagField(
+            self.name,
+            separator=self.separator,
+            case_sensitive=self.case_sensitive,
+            sortable=self.sortable,
+            as_name=self.as_name,
+        )
+
+
+class NumericFieldSchema(BaseField):
+    def as_field(self):
+        return NumericField(self.name, sortable=self.sortable, as_name=self.as_name)
+
+
+class GeoFieldSchema(BaseField):
+    def as_field(self):
+        return GeoField(self.name, sortable=self.sortable, as_name=self.as_name)
+
+
+class BaseVectorField(BaseModel):
+    name: str = Field(...)
+    dims: int = Field(...)
+    algorithm: object = Field(...)
+    datatype: str = Field(default="FLOAT32")
+    distance_metric: str = Field(default="COSINE")
+    initial_cap: Optional[int] = None
+    as_name: Optional[str] = None
+
+    @validator("algorithm", "datatype", "distance_metric", pre=True)
+    def uppercase_strings(cls, v):
+        return v.upper()
+
+    def as_field(self) -> Dict[str, Any]:
+        field_data = {
+            "TYPE": self.datatype,
+            "DIM": self.dims,
+            "DISTANCE_METRIC": self.distance_metric,
+        }
+        if self.initial_cap is not None:  # Only include it if it's set
+            field_data["INITIAL_CAP"] = self.initial_cap
+        return field_data
+
+
+class FlatVectorFieldSchema(BaseVectorField):
```

Comment:
Shouldn't these be kept as just ``xxField``? Fields go into the schema?

Reply:
So the other field types all already had `xxxSchema`, I changed the vector fields to match since they were misnamed in that context. We could strip this from all fields (which makes sense to me).